src/v3/src/transformer/field/attributes.ts

```diff
@@ -30,37 +31,66 @@ const autocompleteValueMap = new Map<string, AutoCompleteValue>([
   ['email', 'email'],
 ]);
 
+const inputModeValueMap = new Map<string, InputModeValue>([
+  ['passcode', 'numeric'],
+  ['totp', 'numeric'],
+  ['email', 'email'],
+  ['phoneNumber', 'tel'],
+]);
+
 const getKeyFromMap = (
-  map: Map<string, AutoCompleteValue>,
+  map: Map<string, AutoCompleteValue> | Map<string, InputModeValue>,
   inputName: string,
 ): string | undefined => {
   let isMatch;
-  map.forEach((value: AutoCompleteValue, key: string) => {
+  map.forEach((_, key: string) => {
     if (inputName.match(key)?.length) {
       isMatch = key;
     }
   });
   return isMatch;
 };
 
-const autocompleteValueTransformer = (input: Input): Nullable<AutoCompleteValue> => {
+const autocompleteValueTransformer = (input: Input): AutoCompleteValue | null => {
   // passcode name is shared with password + totp code types, only differ by secret
   if (input.name === 'credentials.passcode' && input.secret) {
     return autocompleteValueMap.get('password') ?? null;
   }
   const key = getKeyFromMap(autocompleteValueMap, input.name);
+  if (!key) {
+    return null;
+  }
+  const autocompleteValue = autocompleteValueMap.get(key) ?? null;
+  // If not on iOS or Android, disable autocomplete for otp
+  if (autocompleteValue === 'one-time-code' && !isAndroidOrIOS()) {
+    return 'off';
+  }
+  return autocompleteValue;
+};
+
+const inputModeValueTransformer = (input: Input): InputModeValue | null => {
+  // passcode name is shared with password + totp code types, only differ by secret
+  if (input.name === 'credentials.passcode' && input.secret) {
+    return null;
+  }
+  const key = getKeyFromMap(inputModeValueMap, input.name);
   if (key) {
-    return autocompleteValueMap.get(key) ?? null;
+    return inputModeValueMap.get(key) ?? null;
   }
   return null;
 };
 
-export const transformer = (input: Input): Nullable<Result> => {
+export const transformer = (input: Input): Result | null => {
   const attributes: InputAttributes = {};
   const autocompleteValue = autocompleteValueTransformer(input);
   if (autocompleteValue) {
     attributes.autocomplete = autocompleteValue;
   }
+
+  const inputModeValue = inputModeValueTransformer(input);
+  if (inputModeValue) {
+    attributes.inputmode = inputModeValue;
+  }
```

Comment:
To explain why I added `inputmode` https://web.dev/sms-otp-form/#inputmode=numeric